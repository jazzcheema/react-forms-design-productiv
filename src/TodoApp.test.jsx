import { describe, it, expect } from "vitest";

import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoApp from "./TodoApp.jsx";

describe("TodoApp component ", function () {
  const testTodos = [
    {
      title: "Test Title1",
      description: "Test Description1",
      priority: "1",
      id: 1,
    },

    {
      title: "Test Title2",
      description: "Test Description2",
      priority: "2",
      id: 2,
    },

    {
      title: "Test Title3",
      description: "Test Description3",
      priority: "3",
      id: 3,
    },
  ];

  it("Renders without crashing", function () {
    render(<TodoApp initialTodos={testTodos} />);
  });

  it("Matches snapshot", function () {
    const { container } = render(<TodoApp initialTodos={testTodos} />);
    expect(container).toMatchSnapshot();
  });

  it("Removes a todo", function () {
    const { queryAllByText, debug, container } = render(
      <TodoApp initialTodos={testTodos} />
    );
    const deleteBtns = queryAllByText("Del");
    expect(deleteBtns.length).toEqual(3);

    const deleteBtnFire = container.querySelector(".EditableTodo-delBtn");
    fireEvent.click(deleteBtnFire);

    const deleteBtnsAfterDelete = queryAllByText("Del");
    expect(deleteBtnsAfterDelete.length).toEqual(2);
  });

  
  it("Should update a todo", function () {
    const { container } = render(<TodoApp initialTodos={testTodos} />);

    const editBtn = container.querySelector(".EditableTodo-toggle");

    fireEvent.click(editBtn);
    const descriptionArea = container.querySelector("#newTodo-description")

    fireEvent.change(descriptionArea, {target: {value: "Test Change Description"}});
    const submitUpdateBtn = container.querySelector(".NewTodoForm-addBtn")

    fireEvent.click(submitUpdateBtn)
    expect(container).toContainHTML("Test Change Description")
  });
});
