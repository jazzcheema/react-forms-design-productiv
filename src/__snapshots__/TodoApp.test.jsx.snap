// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TodoApp component  > Matches snapshot 1`] = `
<div>
  <main
    class="TodoApp"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div>
          <div
            class="EditableTodo"
          >
            <div
              class="mb-3"
            >
              <div
                class="float-end text-sm-end"
              >
                <button
                  class="EditableTodo-toggle btn-link btn btn-sm"
                >
                  Edit
                </button>
                <button
                  class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                >
                  Del
                </button>
              </div>
              <div
                class="Todo"
              >
                <div>
                  <b>
                    Test Title1
                  </b>
                   
                  <small>
                    (priority: 
                    1
                    )
                  </small>
                </div>
                <div>
                  <small>
                    Test Description1
                  </small>
                </div>
              </div>
            </div>
          </div>
          <div
            class="EditableTodo"
          >
            <div
              class="mb-3"
            >
              <div
                class="float-end text-sm-end"
              >
                <button
                  class="EditableTodo-toggle btn-link btn btn-sm"
                >
                  Edit
                </button>
                <button
                  class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                >
                  Del
                </button>
              </div>
              <div
                class="Todo"
              >
                <div>
                  <b>
                    Test Title2
                  </b>
                   
                  <small>
                    (priority: 
                    2
                    )
                  </small>
                </div>
                <div>
                  <small>
                    Test Description2
                  </small>
                </div>
              </div>
            </div>
          </div>
          <div
            class="EditableTodo"
          >
            <div
              class="mb-3"
            >
              <div
                class="float-end text-sm-end"
              >
                <button
                  class="EditableTodo-toggle btn-link btn btn-sm"
                >
                  Edit
                </button>
                <button
                  class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                >
                  Del
                </button>
              </div>
              <div
                class="Todo"
              >
                <div>
                  <b>
                    Test Title3
                  </b>
                   
                  <small>
                    (priority: 
                    3
                    )
                  </small>
                </div>
                <div>
                  <small>
                    Test Description3
                  </small>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      >
        <section
          class="mb-4"
        >
          <h3>
            Top Todo
          </h3>
          <div
            class="Todo"
          >
            <div>
              <b>
                Test Title1
              </b>
               
              <small>
                (priority: 
                1
                )
              </small>
            </div>
            <div>
              <small>
                Test Description1
              </small>
            </div>
          </div>
        </section>
        <section>
          <h3
            class="mb-3"
          >
            Add Nü
          </h3>
          <form
            class="NewTodoForm"
          >
            <div
              class="mb-3"
            >
              <input
                aria-label="Title"
                class="form-control"
                id="newTodo-title"
                name="title"
                placeholder="Title"
                required=""
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <textarea
                aria-label="Description"
                class="form-control"
                id="newTodo-description"
                name="description"
                placeholder="Description"
              />
            </div>
            <div
              class="mb-3 d-flex justify-content-between"
            >
              <div
                class="w-75 d-flex justify-content-between"
              >
                <label
                  class="d-inline-flex"
                  for="newTodo-priority"
                >
                  Priority:; 
                </label>
                <select
                  class="form-control form-control-sm d-inline-flex"
                  id="newTodo-priority"
                  name="priority"
                >
                  <option
                    value="1"
                  >
                    Ultra-Über
                  </option>
                  <option
                    value="2"
                  >
                    Über
                  </option>
                  <option
                    value="3"
                  >
                    Meh
                  </option>
                </select>
              </div>
              <button
                class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
              >
                Gø!
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
  </main>
</div>
`;
